packages/repository-tests/src/crud/relations/acceptance/has-many-inclusion-resolver.relation.acceptance.ts

```diff
@@ -0,0 +1,273 @@
+            ({customerRepo, orderRepo} = givenBoundCrudRepositories(
+              ctx.dataSource,
+              repositoryClass,
+            ));
+            // inclusionResolvers should be defined. And resolver for each
+            // relation should be created by the hasManyFactory at this point.
+            expect(customerRepo.inclusionResolvers).to.not.be.undefined();
+            expect(orderRepo.inclusionResolvers).to.not.be.undefined();
+            expect(customerRepo.orders.inclusionResolver).to.not.be.undefined();
+            expect(
+              customerRepo.customers.inclusionResolver,
+            ).to.not.be.undefined();
+            // inclusionResolvers shouldn't setup yet at this point
+            expect(customerRepo.inclusionResolvers).to.deepEqual(new Map());
+
+            await ctx.dataSource.automigrate([Customer.name, Order.name]);
+          }),
+        );
+
+        beforeEach(async () => {
+          customerRepo.inclusionResolvers.set(
+            'orders',
+            customerRepo.orders.inclusionResolver,
+          );
+          customerRepo.inclusionResolvers.set(
+            'customers',
+            customerRepo.customers.inclusionResolver,
+          );
+          await customerRepo.deleteAll();
+          await orderRepo.deleteAll();
+        });
+
+        it("defines a repository's inclusionResolvers property", () => {
+          expect(customerRepo.inclusionResolvers).to.not.be.undefined();
+          expect(orderRepo.inclusionResolvers).to.not.be.undefined();
+        });
+
+        it("throws an error if the repository doesn't have such relation names", async () => {
+          await orderRepo.create({
+            customerId: existingCustomerId,
+            description: 'Order from Order McForder, the hoarder of Mordor',
+          });
+          await expect(
+            customerRepo.find({include: [{relation: 'managers'}]}),
+          ).to.be.rejectedWith(
+            `Invalid "filter.include" entries: {"relation":"managers"}`,
+          );
+        });
+
+        it('throws error if the target repository does not have the registered resolver', async () => {
+          await orderRepo.create({
+            customerId: existingCustomerId,
+            description: 'Order from Order McForder, the hoarder of Mordor',
+          });
+          // unregister the resolver
+          customerRepo.inclusionResolvers.delete('orders');
+
+          await expect(
+            customerRepo.find({include: [{relation: 'orders'}]}),
+          ).to.be.rejectedWith(
+            `Invalid "filter.include" entries: {"relation":"orders"}`,
+          );
+          // reset
+          customerRepo.inclusionResolvers.set(
+            'orders',
+            customerRepo.orders.inclusionResolver,
+          );
+        });
+
+        it('simple has-many relation retrieve via find() method', async () => {
+          const c1 = await customerRepo.create({name: 'c1'});
```

Comment:
Since we have a single customer instance only, let's not use counter in the variable name. We also try to avoid single-letter abbreviations because they are often difficult to decipher.

```suggestion
          const customer = await customerRepo.create({name: 'a customer name'});
```